### PR TITLE
add maximum minimum ops

### PR DESCRIPTION
**Pull request by yajiedesign**

### CODE REVIEW

File: src/operator/elementwise_binary_op-inl.h [hunk 1]
```diff
@@ -249,13 +294,15 @@ class ElementWiseBinaryOpProp : public OperatorProperty {
...
     const std::vector<void*> &in_grad) const override {
     switch (GetOpType<ForwardOp>()) {
-      case elembinary::kPlus:
-      case elembinary::kMinus:
-        return {};
-      case elembinary::kMul:
-      case elembinary::kDiv:
-      case elembinary::kPower:
-        return {{out_grad[elembinary::kOut], in_grad[elembinary::kLhs]}};
+    case elembinary::kPlus:
```
Comment (piiswrong) [inline src/operator/elementwise_binary_op-inl.h]: Actually the current design of backward seems strange. Why use template argument for forward op but switch for backward? How about add a template argument BackwardOp? @antinucleon

**Comment by piiswrong** [comment]
LGTM

**Comment by yajiedesign** [comment]
@piiswrong .ok remove mask

**Comment by piiswrong** [comment]
could you do a rebase (git rebase -i HEAD~10) and squash the superfluous commits?

**Comment by yajiedesign** [comment]
@piiswrong  ok

**Comment by piiswrong** [comment]
merged. Thanks!

**Pull request merged**